### PR TITLE
brokers

### PR DESCRIPTION
**Pull request by mabbus**

### CODE REVIEW

File: app/Broker.php [hunk 1]
```diff
@@ -14,22 +14,50 @@ class Broker extends Model
...
 
     protected $casts = [
-        'user_id' => 'integer',
-        'first_name' => 'string',
-        'middle_name' => 'string',
-        'last_name' => 'string',
-        'company' => 'string',
-        'phone' => 'string',
-        'signed_agreement' => 'integer',
+        'user_id' => 'string',
```
Comment (lexod) [inline app/Broker.php:40]: should be integer

File: app/Http/Controllers/Web/EmployeesBrokersController.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    public function list(GetBroker $request) {
+        $data = $request->validated();
+        return response()->json((new GetBrokersTask($data))->run());
+    }
+
+    public function show(GetBroker $request, Broker $broker) {
+        return view('employees.brokers.show', ['title' => 'Business | ' . $broker->first_name, 'broker' => $broker]) ;
+    }
+
+    public function add(Request $request) {
```
Comment (lexod) [inline app/Http/Controllers/Web/EmployeesBrokersController.php:29]: if it belongs to admin - should be AdminBaseRequest

File: app/Http/Requests/Broker/CreateBroker.php [hunk 3]
```diff
@@ -29,15 +29,27 @@ public function authorize()
...
-            'account' => 'required|array',
-            'account.email' => 'required|unique:users,email|string|email|max:191',
-            'profile' => 'required|array',
-            'profile.first_name' => 'required|string|min:2|max:20',
-            'profile.middle_name' => 'nullable|string|min:2|max:20',
-            'profile.last_name' => 'required|string|min:2|max:20',
-            'profile.phone' => 'required|string|max:10',
-            'profile.company' => 'required|max:100',
-            'profile.signed_agreement' => 'required|in:0,1',
+            'owner_first_name' => 'string|min:2|max:20',
```
Comment (lexod) [inline app/Http/Requests/Broker/CreateBroker.php:32]: no required fields?

File: app/Http/Tasks/Broker/CreateBroker.php [hunk 4]
```diff
@@ -5,25 +5,32 @@
...
 
     public function run()
     {
-        $broker = new Broker($this->profileData);
-        $broker->user()->associate($this->account);
-        $broker->save();
+        return User::where('email', $this->data['email'])->firstOr(function ( ){
+            $user = new User($this->data);
+            $user->username = $this->data['email'];
+            $user->password = Hash::make('secret');
```
Comment (lexod) [inline app/Http/Tasks/Broker/CreateBroker.php:25]: take password from data

File: resources/views/layouts/subheader.blade.php [hunk 5]
```diff
@@ -90,6 +90,9 @@ class="kt-badge kt-badge--unified-info kt-badge--inline kt-badge--bold">On Hold<
             @if(strpos(Request::url(), 'syndicates'))
                 <a href="{{ route('employeeSyndicates.add') }}" class="btn btn-brand btn-bold">Add Syndicates</a>
             @endif
+            @if(strpos(Request::url(), 'brokers'))
```
Comment (lexod) [inline resources/views/layouts/subheader.blade.php:93]: check against route name not url

Comment (mabbus) [reply resources/views/layouts/subheader.blade.php:93]: that won't work for multiple URLs

File: app/Http/Tasks/Broker/CreateBroker.php (hunk 4 above)
Comment (mabbus) [reply app/Http/Tasks/Broker/CreateBroker.php:25]: Needs to be revisited.  Automation required

**Pull request merged**